CHANGELOG.md

```diff
@@ -2,6 +2,56 @@
 
 All notable changes to this project are documented in this file.
 
+## 0.23.0
+
+**Release date:** 2022-04-08
+
+This prerelease introduces new retention options for Garbage Collection,
+a new opt-in in-memory cache for `HelmRepository` index files, improves
+notifications following reconciling failures, brings ways to configure 
+Key Exchange Algorithms, plus some extra housekeeping awesomeness.
+
+Garbage Collection is enabled by default, and now its retention options
+are configurable with the flags: `--artifact-retention-ttl` (default: `60s`)
+and `--artifact-retention-records` (default: `2`). They define the minimum
+time to live and the maximum amount of artefacts to survive a collection.
+
+A new notification is now emitted to identify recovery from failures. It 
+is triggered when a failed reconciliation is followed by a successful one, and
+the notification message is the same that's sent in usual successful source
+reconciliation message about the stored artefact.
+
+The opt-in in-memory cache for `HelmRepository` addresses issues where the
+index file is loaded and unmarshalled in concurrent reconciliation resulting
+in a heavy memory footprint. It can be configured using the flags:
+`--helm-cache-max-size`, `--helm-cache-ttl`, `--helm-cache-purge-interval`.
+
+The Key Exchange Algorithms used when establishing SSH connections are
+based on the defaults configured upstream in `go-git` and `golang.org/x/crypto`.
+Now this can be overriden with the flag `--ssh-kex-algos`. Note this only 
+applies to `libgit2` gitImplementation when Managed Transport is being used.
```

Comment:
```suggestion
Now this can be overriden with the flag `--ssh-kex-algos`. Note this applies to the `gogit` gitImplementation or the `libgit2` gitImplementation but _only_ when Managed Transport is being used.
```

The current wording implies that we can only override the kex algos for libgit2 with Managed Transport, but we can also override them for go-git, ref: https://github.com/fluxcd/source-controller/blob/main/pkg/git/gogit/transport.go#L104